Assets/Editor Toolbox/Editor/Hierarchy/HierarchyPropertyLabel.cs

```diff
@@ -282,6 +293,157 @@ public override void OnGui(Rect rect)
             }
         }
 
+        private class HierarchyTreeLinesLabel : HierarchyPropertyLabel
+        {
+            private const float firstElementWidthOffset = 4.0f;
+            private const float firstElementXOffset = -45.0f;
+            private const float startXPosition = 30.0f;
+            private const float columnSize = 14.0f;
+
+            private List<TreeLineLevelRenderer> levelRenderers = new List<TreeLineLevelRenderer>();
+            private int itemRenderCount = 0;
+
+            public override sealed bool UsesWholeItemRect => true;
+
+            public override sealed bool DrawForHeaders => true;
+
+            private bool IsFirstRenderedElement => itemRenderCount == 0;
+
+            public HierarchyTreeLinesLabel()
+            {
+                EditorApplication.update += ResetItemRenderCount;
+            }
+
+            ~HierarchyTreeLinesLabel()
```

Comment:
I'm not sure about the destructor, I have a few other solutions in mind and I'm open for discussion:
- Add Tick method to the HierarchyPropertyLabel (base class) and call it directly in the ToolboxEditorHierarchy for all propertyLabels
- Add IDisposable method to the HierarchyPropertyLabel, call it in the ToolboxEditorHierarchy for all propertyLabels during the static constructor, you can also clear this list